web/src/components/customer/listing-details/contexts/ListingDetailsContext.tsx

```diff
@@ -75,29 +73,44 @@ const ListingDetailsProvider: React.FC<ListingDetailsProviderProps> = ({
   }, [commitStatus, listingId]);
 
   useEffect(() => {
+    if (customer === undefined) {
+      setCommitStatus(undefined);
+      return;
+    }
+
     const fetchCommit = async () => {
       const [commit] = await getCommits({
         listingId,
-        customerId: sampleCustomerId,
+        customerId: customer.id,
       });
       if (commit !== undefined) {
         setCommitId(commit.id);
         setCommitStatus(commit.commitStatus);
       }
     };
     fetchCommit();
-  }, [listingId]);
+  }, [listingId, customer]);
 
   const onCommit = async () => {
+    if (customer === undefined || idToken === undefined) {
```

Comment:
can we just check customer?
is there a case that customer defined but idToken undefined?

Reply:
@ifhuang Logically yes can just check customer! But then we can't just do that because typescript will complain about `idToken` possibly being undefined hahaha